data/blog/how-to-refactor-and-not-break-things.md

```diff
@@ -0,0 +1,101 @@
+---
+title: 'How to Refactor and Not Break Things'
+date: '2024-04-30'
+tags: ['python', 'refactoring', 'sdk']
+draft: false
+summary: 'How we completed a huge refactoring of a software used by thousands of developers without breaking things.'
+images: ['/images/how-to-refactor-and-not-break-things/hero.jpg']
+layout: PostLayout
+authors: ['antonpirker']
+---
+
+In our Python SDK, we completed a huge refactoring, and I want to write down how we pulled this off without breaking (almost) anything and how we managed to stay mostly backward compatible.
+
+## The Initial Situation
+
+When you add the Sentry SDK to your project it hooks into the execution of your application and collects data during execution of your application. Depending on the frameworks or libraries you use the data is collected in different parts of your project and is sent to Sentry in other parts. 
```

Comment:
> it hooks into the execution of your application and collects data during execution of your application.

This is hard to parse. I can't quite give a suggestion though, what are you trying to say exactly?

Something like?
```suggestion
When you add Sentry to your application, it'll instrument the app and runtime to collect useful debugging information at runtime. Depending on the frameworks or libraries you use, the data is collected at different stages of process execution, and is sent to Sentry at a later stage, asynchronously. 
```